### ISSUE TITLE
How to solve the problem which about many 1e-6 value in the gt dist value ?

### ISSUE THREAD

**Issue by cccccccccy** [xieenze/PolarMask#48]
Firstly , thanks for your reply ! When I see your sourse code about generate the gt dist value for  a  mask , I noticed that there are many 1e-6 value because of the missing gt data in this angle range. So I want to know why you can get the 32.9 map with so many 1e-6 data. Thanks again

**Comment by xieenze** [comment]
Hello,
First, for one gt mask, we will sample 9-16 points near the mass center, some points outside the mask will have missing angles(ie 1e-6), but others do not have. 
Second, during training and inference, we have 'polar centerness', which will make the network give high score on high-quality samples,(ie points in the center of the mask), and low score on low-quality samples(ie samples with 1e-6 values).
Third, during inference, all the predicted samples will be processed by NMS, in which samples with a high score will be treat as the final result and the low-quality samples will be removed.

I think the above three analysis can answer your questions.  If you have further questions, please let me know. Thanks.

**Comment by cccccccccy** [comment]
Hello , thanks for your reply. I can understand what you says about generate a good prediction , but i still have a question about a function in your code. “mmdet/dataset/coco_seg.py”
`    def get_36_coordinates(self, c_x, c_y, pos_mask_contour):
        ct = pos_mask_contour[:, 0, :]  # ct 轮廓 各点坐标  size：(num_contour_point,2)
        x = ct[:, 0] - c_x  # 将坐标原点移到 c_X,c_y 处
        y = ct[:, 1] - c_y

        # angle = np.arctan2(x, y)*180/np.pi
        angle = torch.atan2(x, y) * 180 / np.pi  # 获得 轮廓各点 相对坐标原点的 角度
        angle[angle < 0] += 360
        angle = angle.int()

        # dist = np.sqrt(x ** 2 + y ** 2)
        dist = torch.sqrt(x ** 2 + y ** 2) # 获得 轮廓各点 相对坐标原点（c_x,c_y）的 长度len
        angle, idx = torch.sort(angle) # 将 轮廓各点角度 按照大小进行排序
        dist = dist[idx] # 将 轮廓各点长度 按照角度大小排序

        # 生成36个角度
        new_coordinate = {}
        for i in range(0, 360, 10):
            if i in angle:
                d = dist[ angle == i].max()
                new_coordinate[i] = d
            elif i + 1 in angle:
                d = dist[angle == i+1].max()
                new_coordinate[i] = d
            elif i - 1 in angle:
                d = dist[angle == i-1].max()
                new_coordinate[i] = d
            elif i + 2 in angle:
                d = dist[angle == i+2].max()
                new_coordinate[i] = d
            elif i - 2 in angle:
                d = dist[angle == i-2].max()
                new_coordinate[i] = d
            elif i + 3 in angle:
                d = dist[angle == i+3].max()
                new_coordinate[i] = d
            elif i - 3 in angle:
                d = dist[angle == i-3].max()
                new_coordinate[i] = d


        distances = torch.zeros(36)

        for a in range(0, 360, 10):
            if not a in new_coordinate.keys():
                new_coordinate[a] = torch.tensor(1e-6)
                distances[a//10] = 1e-6
            else:
                distances[a//10] = new_coordinate[a]
        # for idx in range(36):
        #     dist = new_coordinate[idx * 10]
        #     distances[idx] = dist

        return distances, new_coordinate`

this function will run after center sample, and it will generate 36 dist for a positive point . but in my opinion, this function still generates a lot of 1e-6 values. Is there something wrong here?
